src/components/productListing/Filters.jsx

```diff
@@ -60,47 +75,75 @@ function Filters() {
           <div>
             <input
               type="checkbox"
-              name="price-range"
-              id="price-50to500"
+              id="under500"
               className="small-text"
+              name="priceCategory"
+              checked={state.priceRange.under500}
+              onChange={() =>
+                dispatch({
+                  type: "PRICERANGE_FILTER",
+                  payload: { priceRange: "under500" },
+                })
+              }
             />
-            <label className="gray-text small-text" htmlFor="item-cakes">
-              ₹ 50 - ₹ 500
+            <label className="gray-text small-text" htmlFor="under500">
+              Under ₹ 500
             </label>
           </div>
 
           <div>
             <input
               type="checkbox"
-              name="price-range"
               id="price-500to1000"
               className="small-text"
+              name="priceCategory"
+              checked={state.priceRange.price500To1000}
+              onChange={() =>
+                dispatch({
+                  type: "PRICERANGE_FILTER",
```

Comment:
instead of writing multiple dispatch functions.

Make a common function, under that call dispatch function. Send args to that function whichever type or payload you want to send.

Example :

  const xxxHandler= (typeOfDispatch, typeOfAction, e) => {
    dataDispatch({
      type: typeOfDispatch,
      payload: typeOfAction,
    });
  };